### PR TITLE
Update jquery to version 2.2.3

### LINKED ISSUE #14634: Bump jQuery to 1.11.2 or 2.x
Can we bump jQuery in Owncloud to 1.11.2 or even better 2.x? The current version used by OwnCloud is not the latest stable and uses deprecated. Notably though 2.x would not work with IE 6/7/8 but maybe thats a good thing as it pushes the web forward? Eitherway though a bump to 1.11.2 would be great in the next maintenance release.

Might aswell also bump any other external libraries to their latest stable.

### PR DESCRIPTION
**Pull request by ChristophWurst**
TODO:
- [x] fix git (sources were removed but not added)
- [x] fix unit tests
- [x] follow http://jquery.com/upgrade-guide/1.9/

fixes #14634

### CODE REVIEW

**Comment by ChristophWurst** [comment]
List of deprecated APIs:
- jQuery.browser
  - [ ] https://github.com/owncloud/core/blob/3ffe1f2e1a88a43687d82e3eedae2c3bb1939aa7/core/js/js.js#L1442

**Comment by ChristophWurst** [comment]
Downgrading to 2.2 as 2.3 causes the following errors:
```
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests file upload dropping external files drop on a tr or crumb outside file list does not trigger upload FAILED
        TypeError: undefined is not a constructor (evaluating 'e.preventDefault()') in /home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js (line 5016)
        preventDefault@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:5016:20
        dispatch@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4741:28
        handle@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4549:33
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7819:17
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery-migrate/jquery-migrate.js:598:26
        triggerHandler@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7909:31
        dropOn@/home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2349:29
        /home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2368:16
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests file upload dropping external files drop on an element outside file list container does not trigger upload FAILED
        TypeError: undefined is not a constructor (evaluating 'e.preventDefault()') in /home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js (line 5016)
        preventDefault@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:5016:20
        dispatch@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4741:28
        handle@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4549:33
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7819:17
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery-migrate/jquery-migrate.js:598:26
        triggerHandler@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7909:31
        dropOn@/home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2349:29
        /home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2378:16
..
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests file upload dropping external files drop on an element inside the table does not trigger upload if no upload permission FAILED
        TypeError: undefined is not a constructor (evaluating 'e.preventDefault()') in /home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js (line 5016)
        preventDefault@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:5016:20
        dispatch@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4741:28
        handle@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4549:33
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7819:17
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery-migrate/jquery-migrate.js:598:26
        triggerHandler@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7909:31
        dropOn@/home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2349:29
        /home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2397:16
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests file upload dropping external files drop on an folder does not trigger upload if no upload permission on that folder FAILED
        TypeError: undefined is not a constructor (evaluating 'e.preventDefault()') in /home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js (line 5016)
        preventDefault@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:5016:20
        dispatch@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4741:28
        handle@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:4549:33
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7819:17
        trigger@/home/christoph/workspace/owncloud/core/vendor/jquery-migrate/jquery-migrate.js:598:26
        triggerHandler@/home/christoph/workspace/owncloud/core/vendor/jquery/dist/jquery.js:7909:31
        dropOn@/home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2349:29
        /home/christoph/workspace/owncloud/apps/files/tests/js/filelistSpec.js:2406:16
```
I assume there's a bug in jQuery, something like https://github.com/jquery/jquery/issues/2917

File: core/js/js.js [hunk 1]
```diff
@@ -629,10 +629,12 @@ var OC={
 		var self = this;
 		$menuEl.addClass('menu');
 		$toggle.on('click.menu', function(event) {
+			console.warn('######');
```
Comment (nickvergessen) [inline core/js/js.js]: hm

File: core/js/js.js [hunk 2]
```diff
@@ -629,10 +629,12 @@ var OC={
...
 			// prevent the link event (append anchor to URL)
 			event.preventDefault();
 
 			if ($menuEl.is(OC._currentMenu)) {
+				console.log('+++++++');
```
Comment (nickvergessen) [inline core/js/js.js]: hm

Comment (ChristophWurst) [inline core/js/js.js]: that was meant for debugging purposes ;-)

**Comment by ChristophWurst** [comment]
@owncloud/javascript have a look please

**Comment by MorrisJobke** [comment]
I tested the basics and also tricky apps like the LDAP wizard - all works fine. I guess we should merge this early to spot all the stuff that is maybe broken due to some legacy usages.

👍 from my side

cc @LukasReschke @nickvergessen @rullzer @blizzz

**Comment by PVince81** [comment]
Yes, let's move this forward. :+1: 

I'd like to have @DeepDiver1975's green light too before merging.

**Comment by PVince81** [comment]
@ChristophWurst once this is merged, please send an email to the dev mailing list to inform app devs, in case they need to make adjustments.

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**